### PR TITLE
Passing experimental: true to D1

### PR DESCRIPTION
**Pull request by geelen**
Fixes # [insert GH or internal issue number(s)].

**What this PR solves / how to test:**

**Associated docs issue(s)/PR(s):**

- [insert associated docs issue(s)/PR(s)]

**Author has included the following, where applicable:**

- [ ] Tests
- [ ] Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))

**Reviewer has performed the following, where applicable:**

- [ ] Checked for inclusion of relevant tests
- [ ] Checked for inclusion of a relevant changeset
- [ ] Checked for creation of associated docs updates
- [ ] Manually pulled down the changes and spot-tested

### CODE REVIEW

File: packages/wrangler/src/d1/create.tsx [hunk 1]
```diff
@@ -19,12 +19,17 @@ export function Options(yargs: CommonYargsArgv) {
 			type: "string",
 			demandOption: true,
 		})
+		.option("experimental", {
```
Comment (penalosa) [inline packages/wrangler/src/d1/create.tsx:22]: Can this have a more verbose name? The word `experimental` is pretty widely used—something like `--experimental-backend` would fit better with the precedent of e.g. `--experimental-local`

Comment (rozenmd) [reply packages/wrangler/src/d1/create.tsx:22]: @penalosa this is for internal testing only

Comment (penalosa) [reply packages/wrangler/src/d1/create.tsx:22]: In that case an environment variable would probably fit better? `EXPERIMENTAL_D1_BACKEND` or something?

Comment (rozenmd) [reply packages/wrangler/src/d1/create.tsx:22]: Nah as in I don't even think we'll merge this into main for now

Comment (penalosa) [reply packages/wrangler/src/d1/create.tsx:28]: Oh right, I didn't realise the base branch wasn't `main`. In that case fire away

**Pull request merged**